File: DXDAL/MoMachines.vb
Imports DevExpress.Data.Filtering
Imports DevExpress.Xpo
Namespace SPGData
    <Persistent("tblMoMachineLine")>
    Public Class MoMachineLine
        Inherits AuditableBase
        <Association("MO-MachineLines")>
        Public Property Mo As MO
        Public Property MachineLine As MachineLine
        Public Property StartTime As Date?
        Public Property EndTime As Date?

        Public Sub New()
            MyBase.New()
        End Sub
        Public Sub New(session As Session)
            MyBase.New(session)
        End Sub

        Public Shadows Class FieldsClass
            Inherits PersistentBase.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property Oid() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Oid"))
                End Get
            End Property
            Public ReadOnly Property MO() As MO.FieldsClass
                Get
                    Return New MO.FieldsClass(GetNestedName("MO"))
                End Get
            End Property
            Public ReadOnly Property MachineLine() As MachineLine.FieldsClass
                Get
                    Return New MachineLine.FieldsClass(GetNestedName("MachineLine"))
                End Get
            End Property
            Public ReadOnly Property StartTime() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("StartTime"))
                End Get
            End Property
            Public ReadOnly Property EndTime() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("EndTime"))
                End Get
            End Property
        End Class
    End Class
End Namespace

File: DXDAL/MoMachines.cs
using System;
using DevExpress.Data.Filtering;
using DevExpress.Xpo;

namespace DXDAL
{
    namespace SPGData
    {
        [Persistent("tblMoMachineLine")]
        public class MoMachineLine : AuditableBase
        {
            [Association("MO-MachineLines")]
            public MO Mo { get; set; }
            public MachineLine MachineLine { get; set; }
            public DateTime? StartTime { get; set; }
            public DateTime? EndTime { get; set; }

            public MoMachineLine() : base()
            {
            }
            public MoMachineLine(Session session) : base(session)
            {
            }

            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty Oid => new OperandProperty(GetNestedName("Oid"));
                public MO.FieldsClass Mo => new MO.FieldsClass(GetNestedName("Mo"));
                public MachineLine.FieldsClass MachineLine => new MachineLine.FieldsClass(GetNestedName("MachineLine"));
                public OperandProperty StartTime => new OperandProperty(GetNestedName("StartTime"));
                public OperandProperty EndTime => new OperandProperty(GetNestedName("EndTime"));
            }
            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
        }
    }
}
